src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -832,28 +828,35 @@ private static ILogger GetLogger(HttpContext httpContext)
             }
         }
 
-        private static void EnsureRequestTaskOfNotNull<T>(Task<T?> task)
+        private static IResult EnsureRequestTaskOfNotNull<T>(Task<T?> task) where T : IResult
         {
             if (task is null)
             {
-                var message = $"Task<{typeof(T).Name}> response should not be null";
-                throw new InvalidOperationException(message);
+                throw new InvalidOperationException("The Task returned by the Delegate must not be null.");
+            }
+
+            var result = task.GetAwaiter().GetResult();
```

Comment:
I'm confused here. You need to make sure the Task is awaited so you don't end up blocking right? There are 2 null checks for `Task<IResult>` and `Task<T> where T : IResult`:

- You check if the `Task<IResult>` itself is null.
- Then you await the `Task<IResult>` to get the IResult and check if the IResult is null before you call ExecuteAsync.

You shouldn't be getting the IResult from the Task using `GetAwaiter().GetResult()` unless you've checked to see if that task is already completed.

Reply:
In fact, the purpose was to check if the **`Task<T> where T: IResult`** is not null and return the underlying IResult (also checked) and use it. You're right, I've missed the **`Task.IsCompletedSuccessfully`** here 😅